### DISCUSSION TITLE
Time stamp in Postal GUI

### DISCUSSION THREAD

**Discussion by eriktroost070** [postalserver/postal#1806] [Help with using Postal]
Recently noticed that the time shown in the message in the Postal GUI is off by exactly 1 hour. It looks like daylight savings time/summer time is still applied.

On the host where Postal is running the correct time is applied, and the hosts sending the email through Postal also have the correct time. Emails received which went through Postal also show the correct time.

So it is only in Postal GUI itself. It's not a big issue but could be confusing when troubleshooting.
Time zone in the settings of the GUI show the correct timezone I'm in.

**Comment by willpower232** [comment]
If you click on `My Settings` in the top right corner, you should be able to set your timezone which I believe will resolve this for you.

![image](https://user-images.githubusercontent.com/1619102/148806161-43259eb6-49fb-4b0d-9599-3bc2c394ae67.png)

**Comment by eriktroost070** [comment]
That is what i thought, but it is set the correct timezone already:
![postal1](https://user-images.githubusercontent.com/74655015/148822196-631412bf-cdbb-4efe-9c57-c2bd7f002834.png)

Time of the most recent send email is in the future:
![postal2](https://user-images.githubusercontent.com/74655015/148822268-18c3352a-cb9d-4444-a28f-c9d06214dd39.png)

Time on the server where Postal is running, which is correct for our timezone and where this server is located.:
![postal3](https://user-images.githubusercontent.com/74655015/148822308-8383b500-1a76-408b-b365-c96a3fe4234b.png)

It looks like it is on the GMT +2 timezone instead of +1.

If I set the timezone to GMT London, then the time is displayed correct for the send messages.

Going to check the containers time/timezone  tomorrow.

Update:
In the container itself it shows the GMT timezone is set.
Rolled out another mail relay with timezone set for the container to Europe/Amsterdam. Tested a few emails, but the messages still show up as if they are send in the future. Container timezone does not seem to influence this.

**Reply by willpower232** [reply]
thanks for the updates, I was going to try different OS timezones but don't quite have the resources at hand at the moment, it works perfectly on ours, even if I change my own user timezone forwards.

It looks like there is some handling of timezones when storing the timestamp in the database https://github.com/postalserver/postal/blob/a735219783127cd0495246ccee89422069da1a1c/lib/postal/message_db/message.rb#L100

Are you able to confirm that the value of timestamp in your messages table is correct?

I have `1642175717.129644` which corresponds to `2022-01-14 15:55:17` UTC which is when I sent the message. If you use the send message page in Postal then it helpfully writes the current time in the subject.

**Reply by eriktroost070** [reply]
I've checked the timestamp on the send message, it is the same incorrect time displayed
